src/java.base/share/classes/java/io/ObjectInputFilter.java

```diff
@@ -118,6 +294,140 @@
+     * The filter returns {@code ALLOWED} or the {@code otherStatus} based on the predicate
+     * of the {@code non-null} class and {@code UNDECIDED} if the class is {@code null}.
+     *
+     * <p>When the filter's {@link ObjectInputFilter#checkInput checkInput(info)} method is invoked,
+     * the predicate is applied to the {@link FilterInfo#serialClass() info.serialClass()},
+     * the return Status is:
+     * <ul>
+     *     <li>{@link Status#UNDECIDED UNDECIDED}, if the {@code serialClass} is {@code null},</li>
+     *     <li>{@link Status#ALLOWED ALLOWED}, if the predicate on the class returns {@code true},</li>
+     *     <li>Otherwise, return {@code otherStatus}.</li>
+     * </ul>
+     * <p>
+     * Example, to create a filter that will allow any class loaded from the platform classloader.
+     * <pre><code>
+     *     ObjectInputFilter f = allowFilter(cl -> cl.getClassLoader() == ClassLoader.getPlatformClassLoader()
+     *                                          || cl.getClassLoader() == null, Status.UNDECIDED);
+     * </code></pre>
+     *
+     * @param predicate a predicate to test a non-null Class, non-null
+     * @param otherStatus a Status to use if the predicate is {@code false}
+     * @return a filter that returns {@code ALLOWED} if the predicate
+     *          on the class is {@code true}
+     * @since 17
+     */
+    static ObjectInputFilter allowFilter(Predicate<Class<?>> predicate, Status otherStatus) {
+        Objects.requireNonNull(predicate, "predicate");
+        Objects.requireNonNull(otherStatus, "otherStatus");
+        return new Config.PredicateFilter(predicate, ALLOWED, otherStatus);
+    }
+
+    /**
+     * Returns a filter that returns {@code Status.REJECTED} if the predicate
+     * on the class is {@code true}.
+     * The filter returns {@code REJECTED} or the {@code otherStatus} based on the predicate
+     * of the {@code non-null} class and {@code UNDECIDED} if the class is {@code null}.
+     *
+     * When the filter's {@link ObjectInputFilter#checkInput checkInput(info)} method is invoked,
+     * the predicate is applied to the {@link FilterInfo#serialClass() serialClass()},
+     * the return Status is:
+     * <ul>
+     *     <li>{@link Status#UNDECIDED UNDECIDED}, if the {@code serialClass} is {@code null},</li>
+     *     <li>{@link Status#REJECTED REJECTED}, if the predicate on the class returns {@code true},</li>
+     *     <li>Otherwise, return {@code otherStatus}.</li>
+     * </ul>
+     * <p>
+     * Example, to create a filter that will reject any class loaded from the application classloader.
+     * <pre><code>
+     *     ObjectInputFilter f = rejectFilter(cl ->
+     *          cl.getClassLoader() == ClassLoader.ClassLoader.getSystemClassLoader(), Status.UNDECIDED);
+     * </code></pre>
+     *
+     * @param predicate a predicate to test a non-null Class, non-null
+     * @param otherStatus a Status to use if the predicate is {@code false}
```

Comment:
At the end of FIlterInputStream javadoc, there is the blanket statement.
```
 * Unless otherwise noted, passing a {@code null} argument to a
 * method in this interface and its nested classes will cause a
 * {@link NullPointerException} to be thrown.``` 
including non-null on the @ param line reinforces the point